### PR TITLE
VA-IIR-426: Add Uptime/Downtime gate for BenefitsEducation::Service

### PR DESCRIPTION
**Pull request by AdamKing0126**
https://app.zenhub.com/workspaces/va-iir-6508c0bd79e64e0fb5855caf/issues/gh/department-of-veterans-affairs/va-iir/426

This PR is a companion to [15642](https://github.com/department-of-veterans-affairs/vets-api/pull/15642) which created a new service to query Lighthouse for a veteran's Post-9/11 Gi Bill statement of benefits.

The service that Lighthouse calls to retrieve this information has scheduled downtime each night.  This PR adds a check which will only permit the call to Lighthouse during scheduled operating hours.  Otherwise, it will return an error, which will be interpreted by the front end client and display a message to the user.

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES/**NO*** - This code is unreachable until we implement a new rails controller which will call it
- *(Summarize the changes that have been made to the platform)* - Create a new service to replace one which calls EVSS
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)* - EVSS is being sunset Sept 2024, so all calls to that service need to be replaced with calls to Lighthouse
- *(Which team do you work for, does your team own the maintenance of this component?)* - I work for VA-IIR and we are the current owner of this product
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- [original PR](https://github.com/department-of-veterans-affairs/vets-api/pull/15642) to create the service.
- [Ticket](https://app.zenhub.com/workspaces/va-iir-6508c0bd79e64e0fb5855caf/issues/gh/department-of-veterans-affairs/va-iir/426)
## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change* - Requests to get GI Bill Statement of Benefits would have been permitted at any time of day.
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing* - executing tests with sandbox credentials loaded.  Verified results.
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?* - We will implement a new rails controller that calls this service.  Testing will be coordinated with front end teams

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
Currently only the newly created service is impacted.

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: config/locales/exceptions.en.yml [hunk 1]
```diff
@@ -2163,7 +2163,13 @@ en:
         <<: *defaults
         code: LH_gateway_timeout
         status: 504
-
+      benefits_education:
+        outside_working_hours:
+          <<: *defaults
+          title: External service unavailable
+          detail: 'Lighthouse GI Bill Status is not available outside of working hours'
+          code: 114
```
Comment (AdamKing0126) [inline config/locales/exceptions.en.yml:2171]: This is the same code # used when raising an exception from EVSS.  I'm assuming it's a good idea to use the same code number.

File: lib/lighthouse/benefits_education/outside_working_hours.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    #
+    def errors
+      [Common::Exceptions::SerializableError.new(i18n_data)]
+    end
+
+    ##
+    # @return [Time] The time to retry the request
+    #
+    def retry_after
+      Time.now.httpdate.in_time_zone('Eastern Time (US & Canada)').to_s
```
Comment (stevelong00) [inline lib/lighthouse/benefits_education/outside_working_hours.rb:22]: Why is this returning the current time?

Comment (AdamKing0126) [reply lib/lighthouse/benefits_education/outside_working_hours.rb:22]: Hey Steve! Good catch, that code's not even used! I copied it from elsewhere 🙄 

The *actual* code to calculate/notify user when they should retry is [here](https://github.com/department-of-veterans-affairs/vets-api/blob/2a4f626813c4978f55767773f53ef9bf6c17c86e/lib/lighthouse/benefits_education/service.rb#L94-L106).  

That code will be utilized to add `Retry-After` headers (coming in the next PR!)

**Review by stevelong00 — APPROVED** [review]
LGTM!

**Review by ryan-mcneil — APPROVED** [review]
👍

**Pull request merged**